api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceServiceImpl.java

```diff
@@ -189,4 +194,25 @@ public Workspace updateUserRoles(Workspace workspace, Set<WorkspaceUserRole> use
     }
     return this.saveWithLastModified(workspace);
   }
+
+  @Override
+  @Transactional
+  public Workspace saveAndCloneCohorts(Workspace from, Workspace to) {
+    // Save the workspace first to allocate an ID.
+    Workspace saved = workspaceDao.save(to);
```

Comment:
FWIW: I think it ought to be possible to cascade inserts on cohorts and reviews such that you don't have to call save() here or on each cohort. But as long as you have @Transactional effectively it's the same thing under the covers I think; this works.

Reply:
Let me get back to this. I've started attempting to test cohort reviews and they don't appear to function as expected - namely I don't understand how they are created (`CreateReview` expects that the review already exists). I also haven't looked at participant annotations; presumably we want to clone those as well. I'm not very familiar with this piece of the data model so I may need a walk-through tomorrow @freemabd1.

Reply:
I think since I'm already doing something custom for participant cohort statuses now, I'll try to use the ORM here.